ssv/runner_signatures.go

```diff
@@ -8,76 +8,71 @@ import (
 	"github.com/ssvlabs/ssv-spec/types"
 )
 
-func (b *BaseRunner) signBeaconObject(
-	runner Runner,
-	obj ssz.HashRoot,
-	slot spec.Slot,
-	domainType spec.DomainType,
-) (*types.PartialSignatureMessage, error) {
+func (b *BaseRunner) signBeaconObject(runner Runner, duty *types.ValidatorDuty,
+	obj ssz.HashRoot, slot spec.Slot, domainType spec.DomainType) (*types.PartialSignatureMessage, error) {
 	epoch := runner.GetBaseRunner().BeaconNetwork.EstimatedEpochAtSlot(slot)
 	domain, err := runner.GetBeaconNode().DomainData(epoch, domainType)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not get beacon domain")
 	}
 
-	sig, r, err := runner.GetSigner().SignBeaconObject(obj, domain, runner.GetBaseRunner().Share.SharePubKey, domainType)
+	sig, r, err := runner.GetSigner().SignBeaconObject(obj, domain,
+		runner.GetBaseRunner().Share[duty.ValidatorIndex].SharePubKey,
+		domainType)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not sign beacon object")
 	}
 
 	return &types.PartialSignatureMessage{
 		PartialSignature: sig,
 		SigningRoot:      r,
-		Signer:           runner.GetBaseRunner().Share.OperatorID,
-	}, nil
-}
-
-func (b *BaseRunner) signPostConsensusMsg(runner Runner, msg *types.PartialSignatureMessages) (*types.SignedPartialSignatureMessage, error) {
-	signature, err := runner.GetSigner().SignRoot(msg, types.PartialSignatureType, b.Share.SharePubKey)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not sign PartialSignatureMessage for PostConsensusContainer")
-	}
-
-	return &types.SignedPartialSignatureMessage{
-		Message:   *msg,
-		Signature: signature,
-		Signer:    b.Share.OperatorID,
+		Signer:           runner.GetOperatorSigner().GetOperatorID(),
+		ValidatorIndex:   duty.ValidatorIndex,
 	}, nil
 }
 
 // Validate message content without verifying signatures
 func (b *BaseRunner) validatePartialSigMsgForSlot(
-	signedMsg *types.SignedPartialSignatureMessage,
+	psigMsgs *types.PartialSignatureMessages,
 	slot spec.Slot,
 ) error {
-	if err := signedMsg.Validate(); err != nil {
-		return errors.Wrap(err, "SignedPartialSignatureMessage invalid")
+	if err := psigMsgs.Validate(); err != nil {
+		return errors.Wrap(err, "PartialSignatureMessages invalid")
 	}
-	if signedMsg.Message.Slot != slot {
+	if psigMsgs.Slot != slot {
 		return errors.New("invalid partial sig slot")
 	}
 
-	// Check if signer is in committee
-	signerInCommittee := false
-	for _, operator := range b.Share.Committee {
-		if operator.OperatorID == signedMsg.Signer {
-			signerInCommittee = true
-			break
+	for _, msg := range psigMsgs.Messages {
+
+		// Check if knows it has the validator index share
+		validatorShare, ok := b.Share[msg.ValidatorIndex]
+		if !ok {
+			return errors.New("unknown validator index")
+		}
+
+		// Check if signer is in committee
+		signerInCommittee := false
+		for _, operator := range validatorShare.Committee {
+			if operator.Signer == msg.Signer {
+				signerInCommittee = true
```

Comment:
you could just return here

Reply:
When we consolidate all messages to use CommitteeID instead of validator pubkey, this hack can be removed

The other solution of leaving validator pubkey would require heavy risky refactoring to spec

Reply:
We actually can't return `nil` if `signerInCommittee = true` here because we are in a for loop and in the next iteration there can still be an error.